plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/AppEngineStandardFacet.java

```diff
@@ -169,19 +173,51 @@ public static void installAppEngineFacet(IFacetedProject facetedProject,
     if (facetedProject.hasProjectFacet(FACET)) {
       return;
     }
+
+    final IFacetedProjectWorkingCopy fpjwc = facetedProject.createWorkingCopy();
+    fpjwc.detect(subMonitor.newChild(20));
+    fpjwc.commitChanges(subMonitor.newChild(20));
```

Comment:
Hmm…  this codepath is the same as that used by WTP's [_Configure > Convert to Faceted Project_](http://git.eclipse.org/c/webtools-common/webtools.common.git/tree/plugins/org.eclipse.wst.common.project.facet.ui/src/org/eclipse/wst/common/project/facet/ui/internal/ConvertProjectToFacetedFormRunnable.java?h=master#n81).  In the absence of some known conflicts, I'm thinking we _should_ use it, and use it more.  More to point, I should change the detectors to also install the AES runtimes too,  and we should use it to install the runtimes in our new project wizards.

I don't think we should get too caught up with the `ConvertJob` fiasco.  Yes it was painful to debug, but in large part because we weren't installing facets atomically, and we were auto-installing the runtime.